geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaFeatureWriter.scala

```diff
@@ -113,12 +145,12 @@ abstract class GeoMesaFeatureWriter[DS <: GeoMesaDataStore[DS, F, W], F <: Wrapp
   override def hasNext: Boolean = false
 
   override def flush(): Unit = {
-    mutators.collect { case c: Flushable => c }.foreach(_.flush())
+    mutators.collect { case c: Flushable => c.flush() }
     statUpdater.flush()
   }
 
   override def close(): Unit = {
-    mutators.collect { case c: Closeable => c }.foreach(_.close())
+    mutators.collect { case c: Closeable => c.close() }
```

Comment:
only thing here with flush and close is the possibility one throws an exception the others won't get called so maybe close quietly...maybe its not too much of an issue...wonder if there is a good pattern for that